jt_notes.md

```diff
@@ -0,0 +1,127 @@
+**I [jonny] have made changes to the source code on a new branch (todos, improved comments, and a few changes to function/variable names etc)**. Where possible I have edited as I go, but am hoping for your input on resolving the queries in the todos. The more important conceptual queries are listed below. **I have also included a preliminary sketch of the architecture I have in mind**, although this is currently limited by my understanding of how the flask stuff works, particularly around multithreading, since that will have a big influence on what a correct and best way to handle all this in python will be. My biggest hangup is that I cannot see any connection between app.py and camera_pi.py (and in particular the frames() iterator). I therefore cannot see how the Flask app stuff is meant to work when running live on a RPi with the PiCam.
+
```

Comment:
camera_pi.py is purely for the live streaming page of the app. The 'live and running with sync' page uses the frame capturing ability that is built in to the raspberry pi's piArray classes. To move away from the pi we will likely need to use openCV (which can be problematic) or similar. They will have two ways of getting frames from a camera: 1) a .start()-.wait()-.stop() approach (like the picam) where we can hopefully extended a method equivalent to .analyze (as we currently do) or 2) a .get_frame() approach, where our own codes could manually get a frame when they're ready. I was thinking that something like: https://www.pyimagesearch.com/2016/01/04/unifying-picamera-and-cv2-videocapture-into-a-single-class-with-opencv/ would help us here.